src/shogun/statistical_testing/internals/DataFetcher.cpp

```diff
@@ -118,7 +118,8 @@ void DataFetcher::shuffle_features()
 			shuffle_subset=SGVector<index_t>(size);
 		}
 		std::iota(shuffle_subset.data(), shuffle_subset.data()+shuffle_subset.size(), 0);
-		CMath::permute(shuffle_subset);
+		// FIXME: Random Refactor PR
+		// CMath::permute(shuffle_subset);
```

Comment:
DataFetcher is not an SGObject, so it can't get a prng object. And this function is virtual, so it can't be templated to accept a PRNG object. The solution I have in mind is to template DataFetcher based on the PRNG type, which I also did with `MaxCrossValidation` (same case)